internal/gatewayapi/runner/runner.go

```diff
@@ -61,15 +61,24 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 		gatewayClasses := r.ProviderResources.GetGatewayClasses()
 		// Fetch the first gateway class since there should be only 1
 		// gateway class linked to this controller
-		t := &gatewayapi.Translator{
-			GatewayClassName: v1beta1.ObjectName(gatewayClasses[0].GetName()),
+		switch {
+		case gatewayClasses == nil:
+			// No need to translate, publish empty IRs, e.g. delete operation.
+			r.XdsIR.Store(r.Name(), nil)
+			r.InfraIR.Store(r.Name(), nil)
+		case gatewayClasses[0] == nil:
+			panic("gatewayclass is nil")
+		default:
+			// Translate and publish IRs.
+			t := &gatewayapi.Translator{
+				GatewayClassName: v1beta1.ObjectName(gatewayClasses[0].GetName()),
+			}
+			// Translate to IR
+			result := t.Translate(&in)
+			// Publish the IR. Use the service name as the key
+			// to ensure there is always one element in the map
+			r.XdsIR.Store(r.Name(), result.XdsIR)
```

Comment:
🙈 I forgot to add the `r.InfraIR.Store(r.Name(), result.InfraIR)` here, I can add it in a future PR if it doesnt get in, in this PR

Reply:
Geez... I added it here in [L68](https://github.com/envoyproxy/gateway/pull/216/files#diff-0a92751feeb3e64824cb22f525cb08b27464d31c38cbeb309bb7f4d9e12e90c0R68) but forgot to add it here. I'll update and push a new commit.

Reply:
@arkodg https://github.com/envoyproxy/gateway/pull/224/ adds `r.InfraIR.Store()` as needed.